Mage.Sets/src/mage/cards/s/SilverScrutiny.java

```diff
@@ -0,0 +1,57 @@
+package mage.cards.s;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.common.CastAsThoughItHadFlashIfConditionAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.dynamicvalue.common.ManacostVariableValue;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.game.Game;
+import mage.game.stack.Spell;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author awjackson
+ */
+public final class SilverScrutiny extends CardImpl {
+
+    public SilverScrutiny(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{X}{U}{U}");
+
+        // You may cast Silver Scrutiny as though it had flash if X is 3 or less.
+        this.addAbility(new CastAsThoughItHadFlashIfConditionAbility(
+                SilverScrutinyCondition.instance,
+                "You may cast {this} as though it had flash if X is 3 or less."
+        ));
+
+        // Draw X cards.
+        this.getSpellAbility().addEffect(new DrawCardSourceControllerEffect(ManacostVariableValue.REGULAR));
+    }
+
+    private SilverScrutiny(final SilverScrutiny card) {
+        super(card);
+    }
+
+    @Override
+    public SilverScrutiny copy() {
+        return new SilverScrutiny(this);
+    }
+}
+
+enum SilverScrutinyCondition implements Condition {
+    instance;
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Spell spell = game.getStack().getSpell(source.getSourceId());
+        if (spell == null) {
+            return false;
+        }
+        return spell.getStackAbility().getManaCostsToPay().getX() < 4;
```

Comment:
How can this condition be checking the X cost on the stack? isn't it too late by the time the spell has been put on the stack?

Reply:
X has already been declared at this point, yes.  The rule modifying effect cancels the spell and shows a message to the player if the X value is too high for casting it with flash.

Reply:
This is literally how rules-modifying effects prohibiting spell casting work, according to 601.2e. A player first puts the spell on the stack, then declaress various choices about how to cast the spell (e.g. value of X, modes, targets), then the proposed spell is checked for legality, and if it is not allowed for whatever reason then the game state is rewound to the moment before the player put the spell on the stack.

Reply:
> This is literally how rules-modifying effects prohibiting spell casting work, according to 601.2e. A player first puts the spell on the stack, then declaress various choices about how to cast the spell (e.g. value of X, modes, targets), then before they start paying costs the proposed spell is checked for legality, and if it is not allowed for whatever reason then the game state is rewound to the moment before the player put the spell on the stack.

Huh. Guess I have more reading to do. I thought that casting a spell only put it on the stack after the cost was paid.